src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs

```diff
@@ -496,12 +500,21 @@ public bool SetUserAgent(string productName, string version)
         /// <returns></returns>
         private string CleanUserAgentInfoEntry(string infoEntry)
         {
-            Regex pattern = new Regex("[~`!@#$%^&*(), ]");            
+            Regex pattern = new Regex("[~`!@#$%^&*(),<>?{} ]");            
             infoEntry = pattern.Replace(infoEntry, "");
 
             return infoEntry;
         }
 
+        private void AddUserAgentEntry(ProductInfoHeaderValue pInfoHeaderValue)
+        {
+            if (!HttpClient.DefaultRequestHeaders.UserAgent.Contains<ProductInfoHeaderValue>(pInfoHeaderValue,
+                new ObjectComparer<ProductInfoHeaderValue>((left, right) => left.Product.Name.Equals(right.Product.Name, StringComparison.OrdinalIgnoreCase))))
```

Comment:
This does not account for UserAgent strings that are comments only.  They will NOT have a .Product property... this will throw a null reference exception.